### PR TITLE
Update dependency management

### LINKED ISSUE #52: Travis checks fail because we've reached Docker pull rate limit without authentication
## Describe the bug
Travis checks fail because we've reached Docker pull rate limit. We need to add Docker authentication information to the Travis config.

## To reproduce
Push commit to remote and see that Travis checks fail due to "toomanyrequests" error. [See example commit](https://app.travis-ci.com/github/RockefellerArchiveCenter/cartographer_backend/builds/232055070).

### PR DESCRIPTION
**Pull request by HaSistrunk**
- Add dependabot config
- Add requirements.in file
- Update pre-commit config to include pip-tools
- Update dependencies
- Adds docker authentication

Fixes #52

### CODE REVIEW

**Comment by helrond** [comment]
I would encourage you to update to Django 3 - if you get stuck on particular things let me know and I can help out.

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ArchivesSnake==0.9.1
+Django==2.2.10
+django-cors-headers==3.7.0
+django-js-asset==1.2.2
```
Comment (helrond) [inline requirements.in]: I believe `django-js-asset` is not a top-level dependency.

File: requirements.in [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+django-mptt==0.12.0
+djangorestframework==3.12.4
+djangorestframework-jwt==1.11.0
+gunicorn==20.1.0
```
Comment (helrond) [inline requirements.in]: I don't think `gunicorn` is actually being used.

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+health-check==3.4.1
+psycopg2-binary==2.9.1
+requests==2.25.1
```
Comment (helrond) [inline requirements.in]: I don't think `requests` is a top-level dependency.

**Review by bonniegee — APPROVED** [review]

File: requirements.in [hunk 4]
```diff
@@ -0,0 +1,10 @@
+ArchivesSnake==0.9.1
+Django==3.2.5
+django-cors-headers==3.7.0
+django-mptt==0.12.0
+djangorestframework==3.12.4
+djangorestframework-jwt==1.11.0
```
Comment (helrond) [inline requirements.in]: I should have caught this earlier, but `djangorestframework-jwt` is not being used in this app.

**Review by helrond — CHANGES_REQUESTED** [review]
One last dependency to remove...

File: requirements.in (hunk 4 above)
Comment (HaSistrunk) [reply requirements.in]: thanks for catching, thought I checked that!

**Review by helrond — APPROVED** [review]

**Review by p-galligan — APPROVED** [review]
Looks good to me!

**Comment by HaSistrunk** [comment]
Checks are failing due to Docker authentication, but passed before the last minimal commit, so I'm going to merge.

**Comment by helrond** [comment]
@HaSistrunk let me push a commit to the Travis file that fixes that.

**Pull request merged**